File: Stack.dart
class Stack {
  int count = 0;
  var storage = {};

  push(val) {
    this.storage[this.count] = val;
    this.count++;
  }

  pop() {
    if (this.count == 0) return null;
    this.count--;
    var result = this.storage[this.count];
    this.storage.remove(this.count);
    return result;
  }

  size() => this.count;

  peak() => this.storage[this.count - 1];
}

void main() {
  final myStack = new Stack();
  myStack.push('Hello');
  print(myStack.peak());
}

File: Stack.js
var Stack = function (){
	this.count = 0;
	this.store = {};

	this.push = (item)=>{
		this.store[this.count] = item ;
		this.count++;
	}

	this.pop =  ()=>{
		(this.count === 0 ? undefined : '');
		this.count --;
		var result = this.store[this.count];
		delete this.store[this.count];
		return result;
		
	}

	this.size = ()=> this.count;

	this.peak = ()=> this.store[this.count-1]
}

var myStack = new Stack();
myStack.push('Hello there!');
myStack.push('Hello there again!');
console.log(myStack.pop());
